tests/workflows/test_embarrassingly_parallel.py

```diff
@@ -0,0 +1,89 @@
+import io
+import tarfile
+import uuid
+
+import coiled
+import pandas as pd
+import pytest
+from dask.distributed import Client, wait
+
+
+@pytest.fixture(scope="module")
+def embarrassingly_parallel_cluster(
+    dask_env_variables,
+    cluster_kwargs,
+    github_cluster_tags,
+):
+    with coiled.Cluster(
+        f"embarrassingly-parallel-{uuid.uuid4().hex[:8]}",
+        environ=dask_env_variables,
+        tags=github_cluster_tags,
+        **cluster_kwargs["embarrassingly_parallel_cluster"],
+    ) as cluster:
+        yield cluster
+
+
+@pytest.fixture
+def embarrassingly_parallel_client(
+    embarrassingly_parallel_cluster,
+    cluster_kwargs,
+    upload_cluster_dump,
+    benchmark_all,
+):
+    n_workers = cluster_kwargs["embarrassingly_parallel_cluster"]["n_workers"]
+    with Client(embarrassingly_parallel_cluster) as client:
+        embarrassingly_parallel_cluster.scale(n_workers)
+        client.wait_for_workers(n_workers)
+        client.restart()
+        with upload_cluster_dump(client), benchmark_all(client):
+            yield client
+
+
+def test_embarassingly_parallel(embarrassingly_parallel_client, s3_factory):
+    # How popular is matplotlib?
+    s3 = s3_factory(requester_pays=True)
+    directories = s3.ls("s3://arxiv/pdf")
```

Comment:
IIUC we'll just let this run on everything that's there. I don't think this is necessary and considering that every month a new directory pops up this would also increase our runtime every month which is bad for a benchmark

Reply:
What if we make a copy of that into our own bucket, like a snapshot in time, and that way we also avoid the whole `request_payer` situation?

Reply:
1991 corresponds to the beginning of the dataset. I think we want to analyze the full dataset (up to some time cutoff so we have a consistent data volume). The 2004/2005 lower bound you're referencing is just for visualization purposes. If you look at the full notebook https://github.com/mrocklin/arxiv-matplotlib/blob/main/arxiv-aws.ipynb you'll see an earlier plot that does back to 1991

Reply:
What I meant to say is that the curve is pretty flat at the beginning and probably including less data for those years would save us some money, if we are running this once a day that is a cost of ~$2 * 365 = ~$730 a year for this workflow only.

Reply:
Ah, I see what you mean. In general, I agree with this sentiment. Though in this particular case, there's not much data before 2004. Comparing back to back runs on 1991-2022 data and 2004-2022 data, I saw a rough price difference of ~$1.66 and ~$1.74, respectively.

Thinking about it more, extending down into the 90s actually has some value as it let's us confirm that the `filename_to_date` utility is working as expected. I've added a few light validation `assert`s to the end of this test. 

We can always revisit the subset of data we use here and frequency we run the workflows in the future if we want to do some price optimization

Reply:
Just checking in @ncclementi does ^ seem reasonable to you?

Reply:
This makes sense, thanks @jrbourbeau 
That was my last comment, I think this is good to go!